### PR TITLE
new branch, now in sync

### PR DESCRIPTION
**Pull request by zacheryconverse**
# Submit a pull request

### 🎯 Goal

Create social app MessageInput

### 🛠 Implementation details

Used much of input from virtual-events app and old social app + new styles

### 🎨 UI Changes

Screen Shot 2021-09-16 at 11 41 24 AM

Screen Shot 2021-09-16 at 11 41 50 AM

Screen Shot 2021-09-16 at 11 42 19 AM

### CODE REVIEW

File: examples/website-demos/social-messaging/src/assets/Attach.tsx [hunk 1]
```diff
@@ -0,0 +1,10 @@
...
+      fill='#72767E'
```
Comment (DanC5) [inline examples/website-demos/social-messaging/src/assets/Attach.tsx:7]: should this color be a variable? you can check with anna

File: examples/website-demos/social-messaging/src/assets/CommandBolt.tsx [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import React from 'react';
+
+export const CommandBolt: React.FC = () => (
+  <svg width='10' height='20' viewBox='0 0 10 20' fill='none' xmlns='http://www.w3.org/2000/svg'>
+    <path
+      d='M4.54545 0H10L6.36364 7.27273H10L3.18182 20L4.54545 10.9091H0L4.54545 0Z'
+      fill='#72767E'
```
Comment (DanC5) [inline examples/website-demos/social-messaging/src/assets/CommandBolt.tsx:7]: should this color be a variable? you can check with anna

File: examples/website-demos/social-messaging/src/assets/EmojiPickerIcon.tsx [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+      fill='#72767E'
```
Comment (DanC5) [inline examples/website-demos/social-messaging/src/assets/EmojiPickerIcon.tsx:9]: should this color be a variable? you can check with anna

File: examples/website-demos/social-messaging/src/assets/GiphyIcon.tsx [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+
+export const GiphyIcon: React.FC = () => (
+  <svg width='70' height='24' viewBox='0 0 70 24' fill='none' xmlns='http://www.w3.org/2000/svg'>
+    <path
+      d='M0 12C0 5.37258 5.37258 0 12 0H58C64.6274 0 70 5.37258 70 12C70 18.6274 64.6274 24 58 24H12C5.37258 24 0 18.6274 0 12Z'
+      fill='#005FFF'
```
Comment (DanC5) [inline examples/website-demos/social-messaging/src/assets/GiphyIcon.tsx:7]: should this color be a variable? you can check with anna

File: examples/website-demos/social-messaging/src/assets/GiphySearch.tsx [hunk 5]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+
+export const GiphySearch: React.FC = () => (
+  <svg width='40' height='40' viewBox='0 0 40 40' fill='none' xmlns='http://www.w3.org/2000/svg'>
+    <path
+      d='M0 20C0 8.95431 8.95431 0 20 0C31.0457 0 40 8.95431 40 20C40 31.0457 31.0457 40 20 40C8.95431 40 0 31.0457 0 20Z'
+      fill='#005FFF'
```
Comment (DanC5) [inline examples/website-demos/social-messaging/src/assets/GiphySearch.tsx:7]: should this color be a variable? you can check with anna

File: examples/website-demos/social-messaging/src/assets/SendArrow.tsx [hunk 6]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+
+export const SendArrow: React.FC = () => (
+  <svg width='40' height='40' viewBox='0 0 40 40' fill='none' xmlns='http://www.w3.org/2000/svg'>
+    <path
+      d='M0 20C0 8.95431 8.95431 0 20 0C31.0457 0 40 8.95431 40 20C40 31.0457 31.0457 40 20 40C8.95431 40 0 31.0457 0 20Z'
+      fill='#B4B7BB'
```
Comment (DanC5) [inline examples/website-demos/social-messaging/src/assets/SendArrow.tsx:7]: should this color be a variable? you can check with anna

File: examples/website-demos/social-messaging/src/assets/index.ts [hunk 7]
```diff
@@ -1,15 +1,23 @@
...
 export { NewChat } from './NewChat';
 export { NewDirectMessage } from './NewDirectMessage';
 export { NewGroupMessage } from './NewGroupMessage';
 export { OptionsIcons } from './OptionsIcons';
 export { SearchIcon } from './SearchIcon';
+export { SendArrow } from './SendArrow';
 export { ThemeIcon } from './ThemeIcon';
+export { EmojiPickerIcon } from './EmojiPickerIcon';
+export { GiphyIcon } from './GiphyIcon';
+export { GiphySearch } from './GiphySearch';
```
Comment (DanC5) [inline examples/website-demos/social-messaging/src/assets/index.ts:23]: abc these

File: examples/website-demos/social-messaging/src/components/Thread/ThreadInner.tsx [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+  const threadOverrideSubmitHandler = useOverrideSubmit(checked);
+
+  return (
+    <>
+      <Thread
+        additionalMessageInputProps={{ overrideSubmitHandler: threadOverrideSubmitHandler }}
+        Input={(props) => (
+          <MessageInput {...props} checked={checked} setChecked={setChecked} threadInput />
+        )}
+        virtualized
```
Comment (DanC5) [inline examples/website-demos/social-messaging/src/components/Thread/ThreadInner.tsx:28]: you shouldn't need a virtualized list in the thread

File: examples/website-demos/social-messaging/src/components/Thread/ThreadInner.tsx [hunk 9]
```diff
@@ -0,0 +1,32 @@
+import React, { useEffect, useState } from 'react';
+import { Thread, useChannelStateContext } from 'stream-chat-react';
+
+import { MessageInput } from '../MessageInput/MessageInput';
+
+import { useOverrideSubmit } from '../../hooks/useOverrideSubmit';
+
+export const ThreadInner = () => {
+  const [checked, setChecked] = useState(false);
```
Comment (DanC5) [inline examples/website-demos/social-messaging/src/components/Thread/ThreadInner.tsx:9]: do the social designs call for a send in channel option? if not we don't need the checked state, this was part of the virtual event app designs

File: examples/website-demos/social-messaging/src/hooks/useOverrideSubmit.ts [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+
+    if (giphyState) {
+      const updatedText = `/giphy ${message.text}`;
+      updatedMessage = { ...message, text: updatedText };
+    }
+
+    const messageToSend = updatedMessage || message;
+
+    try {
+      await sendMessage(messageToSend, { show_in_channel: showInChannel });
```
Comment (DanC5) [inline examples/website-demos/social-messaging/src/hooks/useOverrideSubmit.ts:25]: do we need show in channel? i'm not sure...

File: examples/website-demos/social-messaging/src/components/MessageInput/MessageInput.tsx [hunk 11]
```diff
@@ -0,0 +1,161 @@
...
+      </div>
+      {threadInput && (
+        <div className='thread-footer'>
+          <input
+            checked={checked}
+            className='thread-footer-checkbox'
+            onChange={onCheckChange}
+            type='checkbox'
+          />
+          <div className='thread-footer-text'>Send also as direct message</div>
```
Comment (DanC5) [inline examples/website-demos/social-messaging/src/components/MessageInput/MessageInput.tsx:156]: same question as others, are we doing send in channel in the designs?

File: examples/website-demos/social-messaging/src/components/Thread/Thread.scss [hunk 12]
```diff
@@ -0,0 +1,76 @@
...
+.messaging.str-chat .str-chat__thread-list {
+  padding: unset;
+
+  > :first-child {
+    .message-ui-options {
+      display: none;
+    }
+  }
+
+  .str-chat__virtual-list {
```
Comment (ambaldwin) [inline examples/website-demos/social-messaging/src/components/Thread/Thread.scss:54]: looks like some virtual-list css that can be removed

File: examples/website-demos/social-messaging/src/components/Thread/ThreadHeader.tsx [hunk 13]
```diff
@@ -0,0 +1,28 @@
...
+import { CloseX, Ellipse } from '../../assets';
+
+export const ThreadHeader: React.FC = () => {
+  const { thread } = useChannelStateContext();
+  const { closeThread } = useChannelActionContext();
+
+  if (!thread) return null;
+
+  return (
+    <div className='dm-header-container'>
```
Comment (ambaldwin) [inline examples/website-demos/social-messaging/src/components/Thread/ThreadHeader.tsx:13]: maybe we can update these css classes with more accurate names? DM doesn't really reflect what it is for the social app.

File: examples/website-demos/social-messaging/src/components/ChatContainer/ChatContainer.tsx [hunk 14]
```diff
@@ -5,12 +5,15 @@ import { ChannelSort, LiteralStringForUnion, StreamChat } from 'stream-chat';
 import { Channel, ChannelList, Chat } from 'stream-chat-react';
 
 import { ChannelContainer } from '../ChannelContainer/ChannelContainer';
+import { MessageInput } from '../MessageInput//MessageInput';
```
Comment (ambaldwin) [inline examples/website-demos/social-messaging/src/components/ChatContainer/ChatContainer.tsx:8]: Can we rename the file to SocialMessageInput to keep with the rest of the project?

File: examples/website-demos/social-messaging/src/components/MessageInput/MessageInput.tsx [hunk 15]
```diff
@@ -0,0 +1,161 @@
...
+  CommandBolt,
+  EmojiPickerIcon,
+  GiphyIcon,
+  GiphySearch,
+  SendArrow,
+} from '../../assets';
+
+import { useGiphyContext } from '../../contexts/GiphyContext';
+
+import './MessageInput.scss';
```
Comment (ambaldwin) [inline examples/website-demos/social-messaging/src/components/MessageInput/MessageInput.tsx:24]: Can we rename the scss file to SocialMessageInput as well, to keep with the conventions of the rest of the project.

File: examples/website-demos/social-messaging/src/components/ChatContainer/ChatContainer.tsx [hunk 16]
```diff
@@ -5,12 +5,15 @@ import { ChannelSort, LiteralStringForUnion, StreamChat } from 'stream-chat';
...
 import { SocialChannelPreview } from '../ChannelPreview/SocialChannelPreview';
 import { SocialEmptyStateIndicator } from '../EmptyStateIndicator/SocialEmptyStateIndicator';
 import { SocialMessage } from '../Message/SocialMessageUI';
 import { SideDrawer } from '../SideDrawer/SideDrawer';
 import { SocialChannelList } from '../SocialChannelList/SocialChannelList';
+import { ThreadHeader } from '../Thread/ThreadHeader';
```
Comment (ambaldwin) [inline examples/website-demos/social-messaging/src/components/ChatContainer/ChatContainer.tsx:14]: Can we also rename this file to SocialThreadHeader to keep with the conventions of the rest of the project?

**Pull request merged**